clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -105,6 +105,15 @@ void CompilerInstance::setTarget(TargetInfo *Value) { Target = Value; }
 void CompilerInstance::setAuxTarget(TargetInfo *Value) { AuxTarget = Value; }
 
 bool CompilerInstance::createTarget() {
+  if (getLangOpts().DecimalFloatingPoint) {
+    if (getInvocation().getTargetOpts().CPU == "x86" ||
+        getInvocation().getTargetOpts().CPU == "x86-64")
+      getInvocation().getTargetOpts().DFPEncoding =
+          TargetOptions::BinaryEncoding::BID;
+    else
+      getInvocation().getTargetOpts().DFPEncoding =
+          TargetOptions::BinaryEncoding::DPD;
+  }
```

Comment:
This should be addressed by the code changes I've been taking forever to finish. There is already code that does similarly to populate `TargetInfo::DecimalFloatEnablementAndMode`; see the assignment to `DecimalFloatEnablementAndMode` in `clang/lib/Basic/Targets/OSTargets.h`.